tests/containers/rootless_podman.pm

```diff
@@ -0,0 +1,65 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test rootless mode on podman.
+# - add a user on the /etc/subuid and /etc/subgid to allow automatically allocation subuid and subgid ranges.
+# - check uids allocated to user (inside the container are mapped on the host)
+# - give read access to the SUSE Customer Center credentials to call zypper from in the container
+#   proper way to add Access Control List is via `setfacl` which is not available so we just do
+#   `chmod` instead!! This grants the current user the required access rights
+# - Test rootless container:
+#   * container is launched with default root user
+#   * container is launched with existing user id
+#   * container is launched with keep-id of the user who run the container
+# - Restore /etc/zypp/credentials.d/ credentials
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+use containers::common;
+use containers::container_images;
+use suse_container_urls 'get_suse_container_urls';
+use version_utils qw(get_os_release);
+
+sub run {
+    my ($self) = @_;
+    my ($image_names, $stable_names) = get_suse_container_urls();
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    my $runtime = "podman";
+
+    install_podman_when_needed($host_distri);
+    allow_selected_insecure_registries(runtime => $runtime);
+    my $user      = $testapi::username;
+    my $check_msg = 'Checking allocation range of user';
+    script_run "grep $user /etc/subuid || echo /etc/subuid has no uid range for $user", output => $check_msg;
+    script_run "grep $user /etc/subgid || echo /etc/subgid has no gid range for $user", output => $check_msg;
+    assert_script_run "usermod --add-subuids 200000-201000 --add-subgids 200000-201000 $user";
+    assert_script_run "grep $user /etc/subuid", fail_message => "subuid range not assigned for $user";
+    assert_script_run "grep $user /etc/subgid", fail_message => "subgid range not assigned for $user";
+    # Workaround instead of
+    # "setfacl -m u:$user:r /etc/zypp/credentials.d/*"
+    assert_script_run "chmod -R 666 /etc/zypp/credentials.d/*";
```

Comment:
Is this really needed? And if so, I think that adding 'read' permission should be enough.

Reply:
The user needs acl permissions. although the acl is not available. and this just is used to overcome that for the testing purpose.

Reply:
can you state what we actually try to check here ?

Reply:
i think the documentation might be a better answer to your questions here. https://documentation.suse.com/sles/15-SP2/single-html/SLES-container/index.html#cha-podman-install.

Reply:
> The user needs acl permissions. although the acl is not available. and this just is used to overcome that for the testing purpose.

Could you elaborate where `acl` package is not available ?

Reply:
it was on the host. the image that we produce with the `create_hdd_autoyast_containers`

Reply:
```
install:~ # zypper -q if -t pattern enhanced_base


Information for pattern enhanced_base:
--------------------------------------
Repository      : SLE-Module-Basesystem15-SP3-Pool
Name            : enhanced_base
Version         : 20200124-7.33
Arch            : x86_64
Vendor          : SUSE LLC <https://www.suse.com/>
Installed       : No
Visible to User : Yes
Summary         : Enhanced Base System
Description     :
    This is the enhanced base runtime system with lots of convenience packages.
Contents        :
    S  | Name                        | Type    | Dependency
    ---+-----------------------------+---------+------------
       | multipath-tools             | package | Required
    i+ | patterns-base-base          | package | Required
       | patterns-base-enhanced_base | package | Required
       | OpenIPMI                    | package | Recommended
    i  | aaa_base-extras             | package | Recommended
       | acl                         | package | Recommended
       | alsa-plugins                | package | Recommended
.
.
.
install:~ # zypper -q in -t pattern enhanced_base

The following 283 NEW packages are going to be installed:
  acl alsa-plugins...
```

We are missing `enhanced_base` pattern.

Reply:
+1 to this . I think it is right way to go

Reply:
Or we can add a pattern to `autoyast` profiles, for instance
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/autoyast_sle15/autoyast_containers_x86_64.xml#L123

Reply:
nice catch @mloviska . but lets do it in another PR. in this case either we can block this until then or merge and improve  later